modeshape-jcr/src/main/java/org/modeshape/jcr/value/binary/MongodbBinaryStore.java

```diff
@@ -192,8 +199,10 @@ public BinaryValue storeValue( InputStream stream,
 
             // store content
             DBCollection content = db.getCollection(key.toString());
-            ChunkOutputStream dbStream = markAsUnused ? new ChunkOutputStream(content, System.currentTimeMillis()) : new ChunkOutputStream(
-                                                                                                                                           content);
+            content.createIndex(DATA_CHUNK_SORT_INDEX);
```

Comment:
> each binary value collection must have its own index, so that's the reason this is called each time. Note that `db.getCollection(key.toString());` will create a new collection for each binary value

Hmm, okay. Not sure whether that's a good idea, but I'm not familiar with the limitations or constraints on the number of MongoDB collections. Besides, that's what we've got.

When I mentioned "done in the background", I was not referring to us doing this asynchronously. Instead, [this document](http://docs.mlab.com/indexing/#background-vs-foreground-index-builds) says that one can add a `background=true` field to the index to have MongoDB do it asynchronously. But, this probably doesn't matter because the time to create an index on a newly-created collection should be negligible.